product_margin_classification/models/product_product.py

```diff
@@ -87,10 +88,17 @@ def _get_margin_info(
                     % (product_name)
                 )
             for tax in sale_taxes.filtered(lambda x: x.price_include):
-                multi *= (100 + tax.amount) / 100.0
+                # this code assumes that taxes having price_include set to
+                # True and of amount_type "percent" have also
+                # include_base_amount set to True, while those of amount_type
+                # "fixed" have include_base_amount set to False.
```

Comment:
Better to write a test and raise a not implemented error, if conditions are not checked.

Also, i'm not sur it will work if there are fixed and percent taxes, in various order.

Dont you think ?

Reply:
yes, i agree with you. i was hesitating to add a check and raise an error, but since the current code doesn't do it (while already assuming the condition on "percent" taxes), i didn’t do it. i will add it. note that it might break current configurations.

about multiple tax combinations, i added some tests to test the most common cases.

Reply:
instead of adding exceptions, i improved the code to handle cases with `include_base_amount` `True` or `False`. what do you think?